fs/io_zenfs.cc

```diff
@@ -496,10 +496,10 @@ IOStatus ZonedWritableFile::Truncate(uint64_t size,
 IOStatus ZonedWritableFile::Fsync(const IOOptions& /*options*/,
                                   IODebugContext* /*dbg*/) {
   IOStatus s;
-  LatencyHistGuard guard(zoneFile_->is_wal_
-                             ? &zoneFile_->GetMetrics()->fg_sync_latency_reporter_
-                             : &zoneFile_->GetMetrics()->bg_sync_latency_reporter_);
-  zoneFile_->GetMetrics()->sync_qps_reporter_.AddCount(1);
+  //LatencyHistGuard guard(zoneFile_->is_wal_
+  //                       ? zoneFile_->GetMetrics()->GetRecord("fg_zenfs_sync_latency")
+  //                       : zoneFile_->GetMetrics()->GetRecord("bg_zenfs_sync_latency"));
+  zoneFile_->GetMetrics()->AddRecord("zenfs_sync_qps", 1);
```

Comment:
From my perspective, all such metrics are on critical paths. By moving to this design, you will copy construct a string every time you need to add a record. I think this might have a significant impact on the performance.

Reply:
...you can see how const string& and const char* differs here

https://godbolt.org/z/zrv6zE35M

Reply:
> 

Using this format is obviously not a good choice, even with unordered_map. 
The main reason for using this approach is to decouple and then add parts of the code to the WD's official source code.
...I have forgotten to include the project background in the report.